### DISCUSSION TITLE
Create link by drag & drop

### DISCUSSION THREAD

**Discussion by aylafiryal** [clientIO/joint#2152] [Q&A] [answered]
### Introduction

I'm new at using JointJS and stuck in the process of creating a function to create a link by drag and drop using UML graph. I've tried using magnet but it doesn't work. Where i went wrong?

This is the magnet validation function
![image](https://user-images.githubusercontent.com/83856972/234630328-f88cea7c-d7cf-4bb7-a248-aced22c58171.png)

And this is my UML diagram
![image](https://user-images.githubusercontent.com/83856972/234630791-f0010dc6-905c-4606-8c9f-a02e8441948c.png)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
What are you trying to do? Are you trying to add ports to an UML shape? If yes, please see the [tutorial on ports](https://resources.jointjs.com/tutorial/ports).

**Reply by aylafiryal** [reply]
I simply want my UML shapes to be able to create a link through drag and drop, just like ports work. But if I use ports, the ports' shapes distract my UML appearance. And also there are many types of links for UML class diagrams, so I also need to create a link condition which one the user wants to create

**Reply by kumilingus** [reply, marked as answer]
1. You just add the `magnet` property to the element  you want to become an active magnet.
https://jsfiddle.net/kumilingus/rvLpf4n8

2. You can add [element tools](https://resources.jointjs.com/tutorial/element-tools) to your elements if you want to give your user choice of what link should be created.
https://jsfiddle.net/kumilingus/34sgtufw

**Reply by aylafiryal** [reply]
This works perfectly for my website. Thank you so much for taking the time to help me with this issue!
